src/main/java/com/limechain/network/protocol/warp/dto/Extrinsic.java

```diff
@@ -0,0 +1,33 @@
+package com.limechain.network.protocol.warp.dto;
+
+import com.limechain.utils.HashUtils;
+import io.emeraldpay.polkaj.scale.ScaleCodecWriter;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import org.apache.tomcat.util.buf.HexUtils;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+@Data
+@AllArgsConstructor
+public class Extrinsic {
+    private byte[] extrinsic;
+
+    @Override
+    public String toString(){
+        return HexUtils.toHexString(extrinsic);
+    }
+
+    public byte[] getHash(){
+        ByteArrayOutputStream buf = new ByteArrayOutputStream();
+        try (ScaleCodecWriter writer = new ScaleCodecWriter(buf)) {
+            writer.writeAsList(extrinsic);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        return HashUtils.hashWithBlake2b(buf.toByteArray());
+    }
+
+}
```

Comment:
Since the extrinsic field isn't being changed you can make it `final` and remove the `@AllArgsConstructor` given that `@Data` contains `@RequiredArgsConstructor`.